File: ga/gpt3_5_result_2024_6/result/1968B_cpp_gpt/43_1968B.cpp
use std::io::{self, prelude::*};

fn main() {
    let stdin = io::stdin();
    let mut iterator = stdin.lock().lines().map(Result::unwrap);

    let t: i32 = iterator.next().unwrap().parse().unwrap();

    for _ in 0..t {
        let inputs: Vec<i32> = iterator.next().unwrap()
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        
        let x = inputs[0];
        let k = inputs[1];
        let a = &iterator.next().unwrap();
        let b = &iterator.next().unwrap();

        let mut ans = 0;
        for i in b.chars() {
            if ans < a.len() && a.chars().nth(ans).unwrap() == i {
                ans += 1;
            }
        }

        println!("{}", ans);
    }
}

File: ga/glm4_result_2024_6/result/1968B_cpp_glm/43_1968B.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut lines = input.lines();

    let t: usize = lines.next().unwrap().parse().unwrap();

    for _ in 0..t {
        let mut line = lines.next().unwrap().split_whitespace();
        let x: usize = line.next().unwrap().parse().unwrap();
        let k: usize = line.next().unwrap().parse().unwrap();
        let a: String = line.next().unwrap().to_string();
        let b: String = line.next().unwrap().to_string();

        let mut ans = 0;
        for i in b.chars() {
            if ans < a.len() && a.chars().nth(ans) == Some(i) {
                ans += 1;
            }
        }

        println!("{}", ans);
    }
}